src/mosquito.cr

```diff
@@ -12,6 +12,7 @@ module Mosquito
 
     setting run_cron_scheduler : Bool = true
     setting run_from : Array(String) = [] of String
+    setting backend : Mosquito::Backend.class = Mosquito::RedisBackend
```

Comment:
We did this same thing in LuckyFlow and I don't really care for it. The problem for me is that when you are forced to pass in a class object instead of an instance, you are unable to provide any configuration api of this thing. Seems like it might have also forced you to do this thing with abstract class level methods

https://github.com/luckyframework/lucky_flow/blob/7f43cb734c6dc346597985728baa33138a10435a/src/lucky_flow.cr#L21

Reply:
Thanks for looking, @matthewmcgarvey! What a great point. Customizing the backend as it is would require either subclassing or monkeypatching. This is kind of an inherited limitation because most of the redis methods (class-level or instance-level) haven't changed, they're now just being ported through the abstraction layer.

A bunch of the logic work in RedisBackend is confused, and it exists at the Class level where it should probably exist at the instance level. The concept of a Backend right now is coupled to the notion of named queues, and thats what happens at the instance level -- an instance of a backend represents a named queue in that backend. That probably means the abstraction isn't as clean as it could be.